docs/repo/property-pages/property-specification.md

```diff
@@ -0,0 +1,220 @@
+- The property defines some metadata values
+  - `DependsOn` (optional) lists properties that may influence this property's values (see [Property Dependencies](#Property-Dependencies))
+  - `VisibilityCondition` (optional) holds an expression that the UI will use to determine whether the property should be visible (see [Visibility Conditions](visibility-conditions.md))
+
+## Property Types
+
+Each property in the system has an associated type. For example, the _Assembly name_ property's value is a string, while the _Output type_ property's value is one of a set of enum values.
+
+In the XAML rule files, properties are specified with an underlying type. These types are [defined by MSBuild](https://github.com/dotnet/msbuild/tree/master/src/Framework/XamlTypes) and are:
+
+- `StringProperty`
+- `StringListProperty`
+- `BoolProperty`
+- `EnumProperty`
+- `DynamicEnumProperty`
+- `IntProperty`
+
+## Property Editors
+
+The Project Properties UI maps each of the above property types to a default editor for the underlying type. This default can be overridden in the property's metadata.
+
+Properties may specify additional metadata to modify and/or configure the editor used in the UI. See [property specification](property-specification.md) for further information.
+
+The UI ships a default editor for each of the available property types. 
+
+### Custom Editors
+
+If a non-standard editor is required for a given property, one may be provided via MEF.
+
+⚠ Editor extensibility is under development, tracked by https://github.com/dotnet/project-system/issues/6895. This section will be updated when the issue is resolved.
+
+Using custom editors requires both specifying the editor to use in the XAML rule file (along with any metadata is consumes), as well as ensuring an editor with corresponding name is exported in the client.
+
+To specify a custom editor, add to the property's `ValueEditors` collection in XAML:
+
+```xml
+<StringProperty Name="MyProperty">
+  <StringProperty.ValueEditors>
+    <ValueEditor EditorType="MyEditor">
+      <ValueEditor.Metadata>
+        <NameValuePair Name="Key" Value="Value" />
+      </ValueEditor.Metadata>
+    </ValueEditor>
+  </StringProperty.ValueEditors>
+</StringProperty>
+```
+
+If the `ValueEditors` collection contains multiple entries, the first one having a matching editor on the client is used. If no matching editors are found, the default property editor for the underlying property type (string, in this example) is used.
+
+To expose an editor, export an instance of `IPropertyEditor`, setting the `Name` metadata on the export to match the property's `ValueEditor` `EditorType`.
+
+Continuing the above example:
+
+```c#
+[Export(typeof(IPropertyEditor))]
+[ExportMetadata("Name", "MyEditor")]
+internal sealed class MyPropertyEditor : IPropertyEditor
+{
+    // ...
+}
+```
+
+The `IPropertyEditor` is quite thoroughly documented. See that documentation for guidance on implementing the interface correctly.
```

Comment:
Might be worth calling out that this is documented in CPS.

Reply:
```suggestion
The `IPropertyEditor` is quite thoroughly documented. See that API documentation for guidance on implementing the interface correctly.

Note that the `IPropertyEditor` class is defined in CPS (Microsoft internal) however the documentation is available via IntelliSense.
```